ADR/0006-log-conventions.md

```diff
@@ -0,0 +1,196 @@
+# 6. Log Conventions
+
+Date started: 2022-11-11
+Date revised: 2023-03-08
+
+## Status
+
+Accepted
+
+## Context
+
+We need log conventions to make our offering easier to operate and maintain. These will also make it possible for us to create queries that cross components in our logs, which supports [STONE-201](https://issues.redhat.com/browse/STONE-201).
+
+## Decision: Log conventions
+
+In our controller logs, we will use structured log messages, partially formatted in JSON with key-value pairs as described below.
```

Comment:
It may be worth mentioning that, since all of our controllers (operators) are based on the controller-runtime framework (e.g. via operator-sdk), and it has an opinionated logging mechanism (utility function in `sigs.k8s.io/controller-runtime/pkg/log`, log frontend is logr, log backend is zap), that this logging mechanism does conform to the prescribed format, and can be used.

For example, using the standard controller-runtime logging mechanism will produce a text string in the prescribed format:
```go
logger "sigs.k8s.io/controller-runtime/pkg/log"

func f() {
    log = logger.FromContext(ctx)

    log.Info("log message", "jsonfield1", "jsonfield1value", etc... )
}
```

Reply:
Incorporated in 5b55e6a39048c7afae1b3e22d6c6137d1481093d